accepted/00102-upgrade-to-salt-3008.md

```diff
@@ -0,0 +1,124 @@
+
+With option 1, we do package and do support all those extensions we decide to have a package for. On the other hand, with option 2, we do only take care of maintaing and supporting the reduced list of builin extensions (actually integrated in the main `python311-salt` package), and then users and customers would need to rely on extending with their desired extensions coming from the community.
+
+## Customizing your Salt package
+
+As described by Salt [here](https://salt.tips/the-great-salt-module-migration/), starting with Salt 3008, both Salt Master and Minion can be extended by using `salt-pip` or `pip.installed` to install the Salt Extensions packages (`saltext-*`). Similar to the extending mechanism we currently have for the Salt Bundle.
+
+If an extension is not existing (as most of the dropped modules are NOT yet migrated to Salt Extensions), they mention to provide them as custom modules in your Salt state tree (file_roots).
+
+As mentioned above, we would be probably providing a main Salt package containing some builtin extensions already there that at least covers basic operations.
+
+## Releasing the new Salt
+
+Salt 3008 won't support Python 3.6, so we cannot use the main Python 3.6 interpreter from SLE15 or Leap15, we need to make it to be based on Python 3.11.
+
+Besides of adapting the Salt specfile to build using Python 3.11, by using singlespec adaptations at https://github.com/openSUSE/salt-packaging/pull/96 (and related PRs), we need to take some actions when it comes to how this new version would be distributed in SLE15 and Leap15:
+
+#### SLE15:
+- A new custom "SLE-Module-Salt" module (inherited from SLE-Module-Python3) will be created for SLE15SP4/5/6/7.
+- The new Salt 3008.x package will be shipped via the new "SLE-Module-Salt" together with its any new dependency which is not part of the "SLE-Module-Python3" module.
+- The old Salt 3006.0 package in SLE15SP4/5/6 will be deprecated in favor of the new version coming in the new "SLE-Module-Salt".
+- The old Salt 3006.0 package will be dropped from SLE15SP7 Basesystem and Server Application modules before SLE15SP7 feature cutoff.
+
+#### Leap15:
+- Python 3.11 is already available in Leap.
+- The new Salt and any new dependencies will get into Leap from SLE.
+
+### New dependencies for Salt 3008.
+
+There are new dependencies that are not available as part of "SLE-Module-Python3", and we will need to take it from Factory or some other SLE/ALP source to include them in the new "SLE-Module-Salt" module.
+
+#### Runtime:
+- `python-networkx` + its dependencies (python-matplotlib, python-pandas, python-scipy, python-FontTools)
+- `python-tornado6` 
+
+#### BuildTime:
+- `python-networkx` -> python-pydot, python-graphviz, graphviz
+- `python-meson-python` -> meson, patchelf
+- `python-pandas` -> python-versioneer-toml
+- `python-scipy` -> python-pythran
+
+A testing build project is here: https://build.suse.de/project/show/home:PSuarezHernandez:new-salt-version
```

Comment:
Could we have this in OBS, so that everyone with access to this repo can take a look?